### DISCUSSION TITLE
issue in serving statis assets

### DISCUSSION THREAD

**Discussion by hamza4522** [chaskiq/chaskiq#557] [General]
an issue in serving static assets with Nginx reverse proxy

**Comment by michelson** [comment]
What is the issue? Can you please provide more details?

**Reply by hamza4522** [reply]
i am deploying chasiq on ubuntu server with Nginx as a web server and i am using Nginx as a reverse proxy. and when I mapped 3000 with any of the ubuntu server port,it gives an error on static assets. the issue  is I am unable to serve static assets

**Reply by michelson** [reply]
Hi please add details on the error (the error trace) , nginx config etc...

btw, if you are going to serve assets via NGINX avoid use RAILS_SERVE_STATIC_FILES on env vars

**Comment by hamza4522** [comment]
server {
	listen 9794 default_server;
	listen [::]:9794 default_server;

	passenger_enabled on; 
	
	root /home/hamza/learninghub/chasiq/public;

	# Add index.php to the list if you are using PHP
	index index.html index.htm index.nginx-debian.html;

	server_name localhost:9794;

	location / {
		# First attempt to serve request as file, then
		# as directory, then fall back to displaying a 404.
#        	try_files $uri @ruby;
	
		#try_files $uri $uri/ =404;
                proxy_pass http://localhost:3000; #whatever port your app runs on
                proxy_redirect off;
  	        proxy_set_header X-Real-IP  $remote_addr;
    		proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
   	 	proxy_set_header Host $http_host;
         #       proxy_http_version 1.1;
       	#	proxy_set_header Upgrade $http_upgrade;
               # proxy_set_header Connection 'upgrade';
	        proxy_set_header X-NginX-Proxy true;

               # proxy_set_header Host $host;
               # proxy_cache_bypass $http_upgrade;
	}
 location ~ ^/(assets|packs)/ {
      gzip_static on;
      expires max;
      add_header Cache-Control public;
      try_files $uri =404;
      error_page 404 /404.html;
    }
	location @ruby {
   	# proxy_pass  http://127.0.0.1:3000;
}
		location ~ ^(?!/rails/).+\.(jpg|jpeg|gif|png|ico|json|txt|xml)/ {
     	 	gzip_static on;
      		expires max;
     	        add_header Cache-Control public;
     		try_files $uri =404;
                 error_page 404 /404.html;
}
		# Turn on Passenger
#	passenger_enabled on;
#	passenger_root /usr/lib/ruby/vendor_ruby/phusion_passenger/locations.ini;
	passenger_ruby /home/hamza/.rbenv/versions/2.7.2/bin/ruby;
}
